istioctl/pkg/install/pre-check.go

```diff
@@ -75,28 +75,45 @@ func installPreCheck(istioNamespaceFlag string, restClientGetter genericclioptio
 	fmt.Fprintf(writer, "\n")
 	fmt.Fprintf(writer, "#2. Kubernetes-version\n")
 	fmt.Fprintf(writer, "-----------------------\n")
-	res, err := checkKubernetesVersion(v)
+	res, err := checkKubernetesVersion(version)
 	if err != nil {
 		errs = multierror.Append(errs, err)
 		fmt.Fprint(writer, err)
 	} else if !res {
-		msg := fmt.Sprintf("The Kubernetes API version: %v is lower than the minimum version: "+minK8SVersion, v)
+		msg := fmt.Sprintf("The Kubernetes API version: %v is lower than the minimum version: "+minK8SVersion, version)
 		errs = multierror.Append(errs, errors.New(msg))
 		fmt.Fprintf(writer, msg+"\n")
 	} else {
-		fmt.Fprintf(writer, "Istio is compatible with Kubernetes: %v.\n", v)
+		fmt.Fprintf(writer, "Istio is compatible with Kubernetes: %v.\n", version)
 	}
 
 	fmt.Fprintf(writer, "\n")
 	fmt.Fprintf(writer, "#3. Istio-existence\n")
 	fmt.Fprintf(writer, "-----------------------\n")
-	_, err = c.getNameSpace(istioNamespaceFlag)
-	if err == nil {
-		msg := fmt.Sprintf("Istio cannot be installed because the Istio namespace '%v' is already in use", istioNamespaceFlag)
+	reportNs := ""
+	installNs := ""
+	if len(nsList) == 0 {
+		_, err = c.getNameSpace(v.istioNamespace)
+		if err == nil {
+			reportNs = v.istioNamespace
+		}
+		installNs = v.istioNamespace
+	} else {
+		for _, ns := range nsList {
+			if _, err = c.getNameSpace(ns); err == nil {
+				reportNs += ns + ","
+			}
+			installNs += ns + ","
+		}
+		reportNs = strings.TrimRight(reportNs, ",")
+		installNs = strings.TrimRight(installNs, ",")
+	}
+	if reportNs != "" {
+		msg := fmt.Sprintf("Istio cannot be installed because the Istio namespace '%v' is already in use", reportNs)
```

Comment:
Why? That is how upgrades work. And in the future we support multiple installs

Reply:
Do we want  multi install in the same namespace? This check is make sure we do not install in the same namespace.

Reply:
Yes. But also even today an upgrade is done by installing into a namespace where istio already exists?

Reply:
This pre-check is only used for a brand-new install and not cover upgrade path.  This is also the original function for pre-check.

Reply:
Why do we need so many commands? I think we have more than kubectl at this point and they do a lot more than us. We could just have a single command that checks the cluster? If istio isn't there, it says it can be installed. If it is there it validates it (to be honest I am not really sure what the value of this is, there are 1000s of other applications that run in kubernetes without any "validation")

Reply:
@howardjohn `linkerd check` does this.  I had a problem yesterday; I was getting 503s, I discovered many of my Istio control plane components were in `Evicted`.  I believe it is valuable to have a `check` / `verify-install` command because a failure in the control plane can cause 503s to manifest in different namespaces than the user expects.  There should be an easy way to test the health of Istio itself.

https://linkerd.io/2/reference/cli/check/